src/NServiceBus.AcceptanceTesting/Support/AssemblyScannerExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace NServiceBus.AcceptanceTesting.Support
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    public static class AssemblyScannerExtensions
+    {
+        /// <summary>
+        /// Filters a list of types (e.g. from <see cref="Hosting.Helpers.AssemblyScanner"/>) to only contain types relevant for a specific test endpoint.
+        /// </summary>
+        public static IList<Type> FilterByTest(this IList<Type> types, EndpointCustomizationConfiguration customizationConfiguration) =>
```

Comment:
Can you explain why this doesn't directly use the `AssemblyScanner`? 
Is the idea that there might be use cases where we want to have specific assembly scanner configuration in tests that deviates from `var assemblyScanner = new AssemblyScanner { ScanFileSystemAssemblies = false };`? If that is the case why not have a custom configuration both in this package?

Reply:
Ì was thinking about it but wasn't sure whether it's clear that all the tests need the same scanning behavior. The `FilterByTest` is more specific to handle filtering logic but the assembly scanning logic could also be moved to a helper method although I think the logic should remain separated.